apm-lambda-extension/main.go

```diff
@@ -137,9 +137,14 @@ func main() {
 						log.Println("funcDone signal received, not processing any more agent data")
 						return
 					case agentData := <-agentDataChannel:
+						if !extension.IsTransportStatusHealthy() {
+							extension.WaitForGracePeriod()
```

Comment:
I just realized that blocking here might not be a good idea as this leads to the `backgroundDataSendWg.Wait()` call at the end to block, too.
Maybe just `return` from the goroutine and put the agent data back to the channel?

Reply:
Another safety measure would be to do a 
```go
if !extension.IsTransportStatusHealthy() {
    return
}
```

right before the `for` loop